packages/mermaid/src/diagrams/quadrant-chart/quadrantDb.ts

```diff
@@ -53,8 +54,61 @@ function setYAxisBottomText(textObj: LexTextObj) {
   quadrantBuilder.setData({ yAxisBottomText: textSanitizer(textObj.text) });
 }
 
-function addPoint(textObj: LexTextObj, x: number, y: number) {
-  quadrantBuilder.addPoints([{ x, y, text: textSanitizer(textObj.text) }]);
+function parseStyles(styles: string[]): StylesObject {
+  const stylesObject: StylesObject = {};
+  if (styles.length !== 0) {
+    for (const item of styles) {
```

Comment:
```suggestion
  for (const style of styles) {
```

No need to check for length when using for-of loops. :)
This will avoid one layer of nesting.